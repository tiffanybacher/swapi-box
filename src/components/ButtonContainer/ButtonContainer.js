import React, { Component } from 'react';

class ButtonContainer extends Component {
  handleClick = (e) => {
    const { name } = e.target;

    this.props.setCardData(name);
  }

  render() {
    return (
      <div className="ButtonContainer">
      <button 
        className="peopleBtn"
        onClick={this.handleClick} 
        name='people'>
        People
      </button>
      <button 
        className="planetsBtn"
        onClick={this.handleClick} 
        name='planets'>
        Planets
      </button>
      <button 
        className="vehiclesBtn"
        onClick={this.handleClick} 
        name='vehicles'>
        Vehicles
      </button>
      </div>
    );
  }
}

export default ButtonContainer;