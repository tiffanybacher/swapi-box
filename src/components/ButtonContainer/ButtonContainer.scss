.ButtonContainer {
  display: flex;
  justify-content: space-around;

  button {
    background: black;
    border: 2px solid black;
    border-radius: 3px;
    color: white;
    font-family: 'Roboto', sans-serif;
    font-size: 1.6em;
    margin: 20px;
    padding: 15px;
    min-width: 150px;
  }
}