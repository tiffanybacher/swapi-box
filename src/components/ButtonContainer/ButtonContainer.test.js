import React from 'react';
import { shallow } from 'enzyme';
import ButtonContainer from './ButtonContainer';

describe('ButtonContainer', () => {
  let wrapper;
  let setCardDataMock;

  beforeEach(() => {
    setCardDataMock = jest.fn();
    wrapper = shallow(
      <ButtonContainer setCardData={setCardDataMock} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleClick', () => {
    // let event;

    // beforeEach(() => {
    //   event = {  }
    // });

    it('should invoke setCardData with the correct params when people btn is clicked', () => {
      const eventMock = { target: {name: 'people'}}

      wrapper.find('.peopleBtn').simulate('click', eventMock);

      expect(setCardDataMock).toHaveBeenCalledWith('people');
    });
    
    it('should invoke setCardData with the correct params when planets btn is clicked', () => {
      const eventMock = { target: {name: 'planets'}}

      wrapper.find('.planetsBtn').simulate('click', eventMock);

      expect(setCardDataMock).toHaveBeenCalledWith('planets');
    });
  
    it('should invoke setCardData with the correct params when vehicles btn is clicked', () => {
      const eventMock = { target: {name: 'vehicles'}}

      wrapper.find('.vehiclesBtn').simulate('click', eventMock);

      expect(setCardDataMock).toHaveBeenCalledWith('vehicles');
    });
  });
});
