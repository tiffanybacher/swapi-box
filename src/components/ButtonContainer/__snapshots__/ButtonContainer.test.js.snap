// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match snapshot 1`] = `
<div
  className="ButtonContainer"
>
  <button
    className="peopleBtn"
    name="people"
    onClick={[Function]}
  >
    People
  </button>
  <button
    className="planetsBtn"
    name="planets"
    onClick={[Function]}
  >
    Planets
  </button>
  <button
    className="vehiclesBtn"
    name="vehicles"
    onClick={[Function]}
  >
    Vehicles
  </button>
</div>
`;
